File: backend-go/internal/modules/system/vo/menu_tree_vo.go
package vo

import (
	"drip-admin/backend-go/internal/common"
)

type MenuTreeVo struct {
	ID             common.Int64String `json:"id"`
	ParentID       common.Int64String `json:"parentId"`
	Name           string             `json:"name"`
	Type           string             `json:"type"`
	Path           *string            `json:"path"`
	Component      *string            `json:"component"`
	PermissionCode *string            `json:"permissionCode"`
	Icon           *string            `json:"icon"`
	Sort           int                `json:"sort"`
	Visible        int                `json:"visible"`
	Status         int                `json:"status"`
	Children       []MenuTreeVo       `json:"children"`
}

File: backend-rust/src/modules/system/vo/menu_tree_vo.rs
use crate::common::I64String;
use serde::Serialize;

#[derive(Debug, Clone, Serialize, utoipa::ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct MenuTreeVo {
    pub id: I64String,
    pub parent_id: I64String,
    pub name: String,
    pub r#type: String,
    pub path: Option<String>,
    pub component: Option<String>,
    pub permission_code: Option<String>,
    pub icon: Option<String>,
    pub sort: i32,
    pub visible: i32,
    pub status: i32,
    #[schema(no_recursion)]
    pub children: Vec<MenuTreeVo>,
}
